### PR TITLE
Updated mixedgroup - group leave/join timer - fixed set someone on fire

### PR DESCRIPTION
**Pull request by Th3Dilli**
- Added functionality so when playing mixedgroups there is no + point gain, for example when you are playing a hunter and a ranger is in your group you wont get points for a ranger kill. but the other way round still works because hunters are not supposed to trust each other. This mechanic also works for gutting someone. This function is configurable in the config

- To force people to stay in the group a timer is added when you leave your group so you would have to stay out of your group before you are able to join again. This function is also configurable in the config

- Fixed when you set some one on fire you will instantly get the kill points.

- Fixed some weather entrys in weather array

### CODE REVIEW

**Comment by BOPMatt** [comment]
The basic idea is to not penalize mixed groups but to rather deny them perks and therefore give players a strong incentive to really play the faction system in a more straight-forward fashion. Running groups with factions originally hostile towards each other effectively disables any gain from resulting kills (except for killing factions/traitors that have no allies by definition - such as hunters and independents).

To clarify:
- groups unaffected by the change would be mixed groups of allied factions (e.g. rangers, nomads and survivalists can form groups and will still gain the same amount of points for killing hunters, outlaws and independents)
- killing independents  and hunters is not affected by this change as they don't have allies and can therefore not be betrayed by definition
- killing traitors of any faction is also unaffected by this change
- point loss in general is also unaffected: killing someone from one's own faction (excluding hunters and indies) will still result in point loss and traitor status as usual

Impact:
- outlaws: having one or more outlaws in the group will exclude everyone else in the group from gaining points for killing outlaws - point loss for outlaw vs. outlaw still in effect
- rangers, nomads, survivalists: having one or more in the group will exclude everyone else (e.g. hunters and outlaws) from gaining points for killing any "friendly" class - point loss for friendly vs. friendly still in effect

Examples:

1. Hunter and ranger grouped:
- Ranger kills an outlaw - points are awarded
- Hunter kills an outlaw - points are awarded
- Ranger kills a ranger/nomad/survivalist - loses points, gets flagged as a traitor
- Hunter kills a ranger/nomad/survivalist - no points gained
- Ranger kills a ranger/nomad/survivalist traitor - points are awarded
- Hunter kills a ranger/nomad/survivalist traitor - points are awarded
- Ranger kills a hunter/indipendent - points are awarded
- Hunter kills a hunter/indipendent - points are awarded

2. Hunter and outlaw grouped:
- Outlaw kills a ranger/nomad/survivalist: points are awarded
- Hunter kills a ranger/nomad/survivalist: points are awarded
- Outlaw kills a ranger/nomad/survivalist/outlaw traitor: points are awarded
- Hunter kills a ranger/nomad/survivalist/outlaw traitor: points are awarded
- Outlaw kills an outlaw: loses points, gets flagged as a traitor
- Hunter kills an outlaw: no points gained

... I think you get the point (pun intended).

TLDR: No penalities, no point loss "by association" - but if you want progression: stick to your own kind/allies.
